src/Client.php

```diff
@@ -888,4 +889,16 @@ public function dropOldPartitions(string $table_name, int $days_ago, int $count_
 
         return $result;
     }
+
+    /**
+     * Build sql expression to be used "as is" in a query
+     *
+     * @param string $expression
+     *
+     * @return \ClickHouseDB\Query\Expression
+     */
+    public function expr($expression)
```

Comment:
Do we need this method? I think that instantiating the Expression class is enough. WDYT?

Reply:
For convenience, mostly.

Reply:
Can you specify the added convenience pls?

`$chc->expr("UUIDStringToNum('{$uuid}')");`

vs

`new Expression("UUIDStringToNum('{$uuid}')");`

I'm just trying to find out whether it can be dropped so `Client` class doesn't unnecessarily grow, it already does too much things if you know what I mean ;)

Reply:
Just made a comparison for two cases:
```php
    new Expression("UUIDStringToNum('{$uuid}')")
    $chc->expr("UUIDStringToNum('{$uuid}')")
```

Case with `expr()` call is slightly shorter, as you can see(and will be for most people IMHO). But, what is more important it's much more clear what kind of _expression_ is used here. If you're reading this code for the first time it will be more clear what's going on. In my opinion ofc :)

One more for `expr()` - it incapsulates what's really going on inside. 
So, on one side - interface for `Client` is changed, on the other side it hides innerworks and makes it's easer to change them later.

Reply:
> Case with expr() call is slightly shorter, as you can see(and will be for most people IMHO)

That's just because abbreviation is used but IMO that's not relevant. You could simply make class `E` and say it's shorter. Abbreviations should not be used if not standardized (like IPAddress -> IP)

> But, what is more important it's much more clear what kind of expression is used here

That's rather determined by namespace -> Query Expression. Client has nothing to do with it I think. Client's responsibility is to make connection to database possible and execute operations.

> it incapsulates what's really going on inside

Encapsulation is done by the `Expression` object already, no need for two ;)

Reply:
BTW some cleanup I've worked on recently https://github.com/smi2/phpClickHouse/pull/102 Trying to clean the API so every class doesn't do everything. The goal is to make public API self-explanatory and not confusing.

Reply:
After some thinking I agree that `expr()` method is not `Client` responsibility. But I still think that `Expression` creation is library responsibility. User can, but should not create `Expression` object directly.

Reply:
I will resolve this conversation, as I agreed with `expr()` removal.